cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,360 @@
+elements defined in the CloudEvents specification are to be represented using
+[Extensible Markup Language (XML)](xml-spec) documents.
+
+* The [Attributes](#2-attributes) section describes the representation and
+data type mappings for CloudEvents context attributes.
+
+* The [Data](#3-data) section defines the container for the data portion of a
+CloudEvent.
+
+* The [Envelope](#4-envelope) section defines an XML element to contain CloudEvent
+context attributes and data.
+
+* The [Batch](#5-xml-batch-format) section describes how multiple CloudEvents
+can be packaged into a single XML element.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Approach
+
+The XML representation used is deliberately verbose in favor of readability
+and deterministic processing.
+
+Preservation of type information for CloudEvent attributes is supported allowing
+custom extensions to be communicated without type loss.
+
+A schema-less approach has been taken favoring convention over rigid document
+structure.
+
+The namespace `http://cloudevents.io/xmlformat/V1` MUST be used.
+
+When namespace prefixes are used a prefix of `ce` is preferred but MUST NOT
+be expected from an XML document processing perspective.
+
+An XML document preamble SHOULD be included to ensure deterministic processing.
+
+## 2. Attributes
+
+The CloudEvents type system is mapped to a set of CloudEvent specific
+type designators as follows :
+
+| CloudEvents Type  |  XML Format Type | XML Schema Type | Notes |
+| :-----------------| :----------------| :--------------- | :---- |
+| Boolean       | ce:boolean | [xs:boolean][xml-primitives] | `true` or `false` |
+| Integer       | ce:integer | [xs:int][xml-primitives] | |
+| String        | ce:string | [xs:string][xml-primitives] | |
+| Binary        | ce:binary | [xs:base64Binary][xml-primitives] |  |
+| URI           | ce:uri | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986]| |
+| URI-reference | ce:uriRef | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986] | |
+| Timestamp     | ce:timestamp | [xs:dateTime][xml-primitives] following [RFC 3339][rfc3339] | |
+
+Each CloudEvent context attribute MUST be represented as an XML element whose local
+name exactly matches that of the attribute.
+
+See the [envelope](#4-envelope) for special handing of the `specversion` context attribute.
+
+Extension context attributes MUST be decorated with the appropriate CloudEvent type format
+designators using an `xsi:type` XML attribute, this allows them to be exchanged
+without loss of type information.
+
+REQUIRED and OPTIONAL context atrtribute MAY be decorated with `xsi:type`, if present this
+value MUST match that of the type specified by the [CloudEvent type specification][ce-types].
```

Comment:
I think you meant to reference the type definition per attribute, not the types table. As currently worded, someone could interpret it as "it must be one of our types", but not necessarily "the type defined for THAT attribute per the spec". Perhaps:
- - -
if present this value MUST match that of the type specified for the specific context attribute in the [CloudEvents content attributes](#context-attributes) section of the specification.
- - -

Reply:
@JemDay ^

Reply:
Ack